crates/chia-datalayer/src/merkle/iterators.rs

```diff
@@ -61,13 +63,39 @@ impl Iterator for LeftChildFirstIterator<'_> {
                 }
             }
 
+            match block.node.parent().0 {
+                // TODO: maybe also check the parent considers this the child?
+                Some(index) => {
+                    if item.index != self.from_index && !self.already_queued.contains(&index) {
+                        return Some(Err(Error::ReferenceToUnknownParent()));
+                    }
+                }
+                None => {
+                    if item.index.0 != 0 {
+                        return Some(Err(Error::UnexpectedParentlessNode()));
```

Comment:
I don't really know what's going on, but I would expect this to happen when you get to the root. Is that not expected?

Reply:
presently the root is at index 0.  we have considered not requiring that, but have not made such a change.  so this checks if the parent is none and the item is not the root (at index 0).